src/test/groovy/edu/ie3/datamodel/models/value/IrradiationValueTest.groovy

```diff
@@ -0,0 +1,30 @@
+/*
+ * © 2020. TU Dortmund University,
+ * Institute of Energy Systems, Energy Efficiency and Energy Economics,
+ * Research group Distribution grid planning and operation
+ */
+package edu.ie3.datamodel.models.value
+
+import edu.ie3.datamodel.models.StandardUnits
+import edu.ie3.util.quantities.PowerSystemUnits
+import spock.lang.Specification
+import tech.units.indriya.quantity.Quantities
+
+
+class IrradiationValueTest extends Specification {
+
+	def "An Irradiation's equal method should work as expected on null and provided quantities"() {
+		expect:
+		(iVal1 == iVal2) == res
+
+		where:
+		iVal1                                                                                                | iVal2                                                                                || res
+		new IrradiationValue(null, null)                                                                     | new IrradiationValue(null, null)                                                     || true
+		new IrradiationValue(Quantities.getQuantity(10d, StandardUnits.IRRADIATION), null)                   | new IrradiationValue(Quantities.getQuantity(10d, StandardUnits.IRRADIATION), null)   || true
+		new IrradiationValue(Quantities.getQuantity(10d, StandardUnits.IRRADIATION), null)                   | new IrradiationValue(null, null)                                                     || false
+		null                                                                                                 | new IrradiationValue(null, null)                                                     || false
+		null                                                                                                 | null                                                                                 || true
+		new IrradiationValue(Quantities.getQuantity(10.23d, StandardUnits.IRRADIATION), null)                | new IrradiationValue(Quantities.getQuantity(10.23, StandardUnits.IRRADIATION), null) || false
+		new IrradiationValue(Quantities.getQuantity(10230, PowerSystemUnits.WATTHOUR_PER_SQUAREMETRE), null) | new IrradiationValue(Quantities.getQuantity(10.23, StandardUnits.IRRADIATION), null) || false
+	}
+}
```

Comment:
Do we need to fill up all other (yet missing) tests as well?

Reply:
Depends - I would say no, bc I think it's not necessary at all. But if you and @mia-krause think it is necessary/make sense I could add them. Just let me know pls.

Reply:
If you want to be really strict about it, they definitely would enforce the nullable equals on weather values. However, I don't think (or hope..) that we will change the equals methods a lot in the future and it also wouldn't enforce the pattern on new entities. So I personally don't think that we need the other tests.